chukwuemekachm/Maintenance-Tracker
# #163355831 Completes the setup for data access layer

#### What does this PR do?
- This PR completes the implementation of the data access layer on the application.
#### Description of Task to be completed?
- Create Data Access Objects for every object in the database
- Write tests for data access objects
- Organize tests into subfolders in the tests directory
#### How should this be manually tested?
- Set up the repo [locally](https://github.com/chukwuemekachm/Maintenance-Tracker#installation)
- Run `npm run test`
- Observe two new test suites have been introduced to the tests.
#### Any background context you want to provide?
- Pre4viously all data access operations were performed in the contollers and there was no distinction between business logic and data access.
- With the new Data Access Layer it will be very easy and efficient to upgrade and change the data storage without refactoring the core business logic of the application.
- It also decouples the system and provides a clear separation of concern.
#### What are the relevant pivotal tracker stories?
- [#163355831](https://www.pivotaltracker.com/story/show/163355831)
#### Screenshots (if appropriate)
- N/A
#### Questions:
- N/A
